app/controllers/index.rb

```diff
@@ -0,0 +1,147 @@
+enable :sessions
+
+get '/' do
+  redirect '/user' if session[:user_id]
+    @da = Station.find(13)
+    @du = Station.find(15)
+    @fr = Station.find(19)
+    @ml = Station.find(26)
+    @pt = Station.find(31)
+    @rd = Station.find(34)
+  @stations = Station.all
+  erb :index
+end
+
+
+post '/eta' do
+  session[:station] = params[:station]
+  session[:destination] = params[:destination]
+  session[:eta] = []
```

Comment:
You are storing too much in the session. HTTP should remain as stateless as possible.